cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/GitHubHttpHandler.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright © 2015-2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.gateway.handlers;
+
+import com.google.gson.Gson;
+import com.google.inject.Inject;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.gateway.handlers.util.AbstractAppFabricHttpHandler;
+import io.cdap.cdap.internal.app.store.GitHubStore;
+import io.cdap.cdap.internal.github.GitHubRepo;
+import io.cdap.cdap.internal.github.GitHubRequest;
+import io.cdap.http.HttpResponder;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.http.HttpResponseStatus;
+
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+
+@Path(Constants.Gateway.API_VERSION_3)
+public class GitHubHttpHandler extends AbstractAppFabricHttpHandler {
+  private static final Gson GSON = new Gson();
+  private final GitHubStore gitStore;
+
+
+  @Inject
+  GitHubHttpHandler(GitHubStore gitStore) {
+    this.gitStore = gitStore;
+  }
+
+  @Path("repos/github")
+  @GET
+  public void getRepos(HttpRequest request, HttpResponder responder) throws Exception {
+    responder.sendJson(HttpResponseStatus.OK, GSON.toJson(gitStore.getRepos()));
+  }
+
+  /**
+   * Returns user repository information
+   */
+  @Path("repos/github/{repo}")
+  @GET
+  public void getRepoInfo(HttpRequest request, HttpResponder responder,
+                          @NotNull @PathParam("repo") String repo) throws Exception {
+    try {
+      responder.sendJson(HttpResponseStatus.OK, GSON.toJson(gitStore.getRepo(repo)));
+    } catch (Exception ex) {
+      responder.sendStatus(HttpResponseStatus.NOT_FOUND);
```

Comment:
not all exception means NOT_FOUND.  Instead of relying on exception, you'd better check whether result is null or an empty Optional